### DISCUSSION TITLE
close a webview tab on profile switch

### DISCUSSION THREAD

**Discussion by sefatanam** [microsoft/vscode-discussions#1346] [Extension Development QnA] [answered]
I'm developing a VS Code extension where I need to clean up resources (specifically webview panels) when the user switches profiles. According to the [official documentation](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#profiling-support), the deactivate function will be called on profile switch, but asynchronous cleanup tasks might not complete before the extension is deactivated. So, I want to close the tab of my extension should close when you switch the profile.

![demo](https://github.com/user-attachments/assets/58494a9b-a098-4b3b-aeef-9b476d920c46)

**Comment by isidorn** [comment]
Ok - I think you are on the right track. On your deactivate you should do your cleanup - like closing the tab of your extension.
What seems to be the issue here? Does your deactivate get called? Your extension does not get enough time?
On deactivate do you get an option to pause the extension host close?

**Reply by sefatanam** [reply]
Yes, you understood my issue. My extension does not get sufficient time to properly dispose. The process executes immediately, preventing the webview's dispose function from being called in time not even get chance to execution.

**Reply by isidorn** [reply]
@bpasero I know we have API to pause close of processes, but that is only internal right? Extensions can not not do this?

**Reply by bpasero** [reply]
We do not expose that as API for extensions. I would ping Matt here too if there is something web view specific that can be done.

**Reply by isidorn** [reply]
@mjbvz in case you have ideas.

**Reply by mjbvz** [reply, marked as answer]
I don't think this is possible for plain webviews but should be possible using a custom editor. If the custom editor is dirty, you should get a chance to save any data before everything is shut down

If you have to use normal webview instead, your best bet may to be to save the data eagerly instead of waiting for shutdown
